src/macports1.0/macports.tcl

```diff
@@ -5219,12 +5213,25 @@ proc macports::revupgrade_buildgraph {port stackname adjlistname revadjlistname
 
 # get cached ping time for host, modified by blacklist and preferred list
 proc macports::get_pingtime {host} {
-    global macports::ping_cache macports::host_blacklisted macports::host_preferred
-    if {[info exists host_blacklisted($host)]} {
-        return -1
-    } elseif {[info exists host_preferred($host)]} {
-        return 1
-    } elseif {[info exists ping_cache($host)]} {
+    global macports::ping_cache macports::host_cache \
+           macports::host_blacklist macports::preferred_hosts
+
+    if {[info exists host_cache($host)]} {
+        return $host_cache($host)
+    }
+    foreach pattern $macports::host_blacklist {
+        if {[string match $pattern $host]} {
```

Comment:
We should probably make this `string match -nocase`.

Reply:
Sounds reasonable.